### ISSUE TITLE
Cannot run Rails server after curb 0.9.3 install in Windows 10

### ISSUE THREAD

**Issue by maxence33** [taf2/curb#321]
I get the following error when starting rails server (puma) after curb 0.9.3 install

`C:/Ruby23/lib/ruby/gems/2.3.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in require: 126: Le module spÚcifiÚ est introuvable.   - C:/Ruby23/lib/ruby/gems/2.3.0/gems/curb-0.9.3/lib/curb_core.so (LoadError)`

I struggled to install curb and curl. But basically I did this : 

`gem install curb -- --platform=ruby -- -- --with-curl-lib="C:/curl-7.54.1-win32-mingw/bin" --with-curl-include="C:/curl-7.54.1-win32-mingw/include"`

And the curb install went smoothly.
I have also included the libcurl.dll to Ruby/bin folder. 

I am not sure where the problem come from. I have struggled so much to install curb and curl that it may be the problem. I have found 2 versions of Devkit on my Platform. One 32 bits and one 64 bits. But I guess I would'nt have been able to install curb in case of mismatch ...

I have also seen extensive closed issues about install and extensive answers (like this one https://github.com/taf2/curb/issues/315 ) but I am not sure I should dive in an attempt to reinstall curb if the problem is not related to curb install ...

(if someone suspects a specific error, I am happy to follow another closed issue answers without a full tutorial here)

**Comment by taf2** [comment]
Can you try loading from a irb shell... with maybe:

require 'curb'

Stack trace with that might be more revealing?   Also does windows have a tool like otool -L or ldd on linux to check the load paths of the compiled dll?

**Comment by maxence33** [comment]
seems the same error : 

irb(main):001:0> require 'curb'
LoadError: 126: Le module spÚcifiÚ est introuvable.   - C:/Ruby23/lib/ruby/gems/2.3.0/gems/curb-0.9.3/lib/curb_core.so
        from C:/Ruby23/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from C:/Ruby23/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from C:/Ruby23/lib/ruby/gems/2.3.0/gems/curb-0.9.3/lib/curl.rb:1:in `<top (required)>'
        from C:/Ruby23/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from C:/Ruby23/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from C:/Ruby23/lib/ruby/gems/2.3.0/gems/curb-0.9.3/lib/curb.rb:1:in `<top (required)>'
        from C:/Ruby23/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:127:in `require'
        from C:/Ruby23/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:127:in `rescue in require'
        from C:/Ruby23/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:40:in `require'
        from (irb):1
        from C:/Ruby23/bin/irb.cmd:19:in `<main>'
irb(main):002:0>

Also I have not installed a development version of curl. I will try with a devel version if I can find one. 
Also maybe installing a slightly older version of curb. It is the 0.9.3 version that is installed by `zipline` but I will install another version and add it to my gemfile and see what happens.

Also I don't have anything like otool. In last resort I will try install from #315 thread.

**Comment by maxence33** [comment]
Hi Todd,

I have tried a few things. 

first : 
gotten rid of my extra DEVKIT versions. 
Made sure Devkit-mingw64-32-4.7.2 was properly installed with paths etc ...
gotten rid of all lcurl versions 
installed curl-7.55.0-win32-mingw with paths
installed curb gem manually with : 
gem install curb -- --platform=ruby -- -- --with-curl-lib="C:/curl-7.55.0-win32-mingw/bin" --with-curl-include="C:/curl-7.55.0-win32-mingw/include"
gem curb 0.9.3 installed successfully 
yet still same error

second: 
I followed thread 315 and compiled my own libcurl Library with my own Devkit version.
made sure libcurl-4.dll was located in folder : C:\Ruby23\lib\ruby\gems\2.3.0\gems\curb-0.9.3\ext
Still same error

Yet I saw on stackoverflow people adding the libs directly to Ruby/bin folder. Then I copied libcurl-4.dll and libcurl.dll (same gem as former but renamed)  in Ruby/bin folder and All successful. 
**Puma running like a charm** 

I think #315 is the last resort install with adding the newly created libs both to Ruby/bin and gem folder too.

(still got a Mime Type error but I think it is because a line I added in the Mime initializers for Wicked_pdf. Once removed, all good. So different problem)
